src/cpp/rtps/transport/TCPTransportInterface.h

```diff
@@ -237,9 +240,9 @@ class TCPTransportInterface : public TransportInterface
     bool CloseInputChannel(
             const Locator&) override;
 
-    //! Removes all outbound sockets on the given port.
+    //! Resets the locator bound to the sender resource.
     void CloseOutputChannel(
-            std::shared_ptr<TCPChannelResource>& channel);
+            fastrtps::rtps::Locator_t& locator);
```

Comment:
This function receives the attribute _locator__ of the sender resource to set its listening port to 0, so it needs to be modified within the function. However, it could be deleted as the locator will be deleted by the default destructor.

Reply:
Let's leave it as it is now, then